clickhouse/types/types.h

```diff
@@ -1,13 +1,16 @@
 #pragma once
 
-#include "absl/numeric/int128.h"
-
 #include <map>
 #include <memory>
 #include <string>
 #include <vector>
 #include <stdexcept>
 
+namespace absl
+{
+class int128;
+}
+
```

Comment:
Isn't it better to move all `Int128` specific code and support into `clickhouse/types/int128.h` so that when user doesn't include that header he can't even refer to any type related/implemented via `Int128`?